service-base/src/main/java/org/eclipse/hono/service/AbstractProtocolAdapterBase.java

```diff
@@ -485,38 +485,33 @@ protected void doStop(final Future<Void> stopFuture) {
      * Create a command response sender for a specific device.
      *
      * @param tenantId The tenant of the command receiver.
-     * @param deviceId The device of the command receiver.
      * @param replyId The replyId to from the command to use for the response.
      * @return Result of the response sender creation.
      */
     public final Future<CommandResponseSender> createCommandResponseSender(
             final String tenantId,
-            final String deviceId,
             final String replyId) {
-        return commandConnection.getOrCreateCommandResponseSender(tenantId, deviceId, replyId);
+        return commandConnection.getOrCreateCommandResponseSender(tenantId, replyId);
     }
 
     /**
      * Forwards a response message that has been sent by a device in reply to a
      * command to the sender of the command.
      * 
      * @param tenantId The tenant that the device belongs to.
-     * @param deviceId The identifier of the device.
      * @param response The response message.
      * @return A future indicating the outcome of the attempt to send
      *         the message.
      * @throws NullPointerException if any of the parameters are {@code null}.
      */
     protected Future<ProtonDelivery> sendCommandResponse(
             final String tenantId,
-            final String deviceId,
             final CommandResponse response) {
 
         Objects.requireNonNull(tenantId);
-        Objects.requireNonNull(deviceId);
         Objects.requireNonNull(response);
 
-        return createCommandResponseSender(tenantId, deviceId, response.getReplyToId())
+        return createCommandResponseSender(tenantId, response.getReplyToId())
```

Comment:
wouldn't it make sense to add a *sendCommandResponse(CommandResponse response)* method to `CommandResponseSender`?

Reply:
Since only the tenantId and replyId are needed for this I think it is clearer to stay with this signature -also only the CommandResponse would not offer the tenantId.

Reply:
??? I am talking about *CommandResponseSender.sendCommandResponse()*

Reply:
Ah, I see - yes, that makes sense.